prow/git/v2/client_factory.go

```diff
@@ -387,16 +387,19 @@ func (c *clientFactory) ClientForWithRepoOpts(org, repo string, repoOpts RepoOpt
 
 	// First create or update the primary clone (in "cacheDir").
 	timeBeforeEnsureFreshPrimary := time.Now()
-	c.ensureFreshPrimary(cacheDir, cacheClientCacher, repoOpts, org, repo)
-	gitMetrics.ensureFreshPrimaryDuration.WithLabelValues(org, repo).Observe((float64(time.Since(timeBeforeEnsureFreshPrimary).Seconds())))
+	err = c.ensureFreshPrimary(cacheDir, cacheClientCacher, repoOpts, org, repo)
+	if err != nil {
+		c.logger.WithFields(logrus.Fields{"org": org, "repo": repo, "dir": cacheDir}).Errorf("Ensure fresh primary clone failed with error: %s", err.Error())
+	}
+	gitMetrics.ensureFreshPrimaryDuration.WithLabelValues(org, repo).Observe(time.Since(timeBeforeEnsureFreshPrimary).Seconds())
 
 	// Initialize the new derivative repo (secondary clone) from the primary
 	// clone. This is a local clone operation.
 	timeBeforeSecondaryClone := time.Now()
 	if err := repoClientCloner.CloneWithRepoOpts(cacheDir, repoOpts); err != nil {
 		return nil, err
 	}
-	gitMetrics.secondaryCloneDuration.WithLabelValues(org, repo).Observe((float64(time.Since(timeBeforeSecondaryClone).Seconds())))
```

Comment:
Why is the float64 cast here being removed? IIRC I added this metric and when I did so, I just copied the existing callers to Observe().

Reply:
The conversion is not necessary, since the `Seconds` method returns a float64.

Reply:
Can you check if this is true for the version of the Prometheus library we are using?

Reply:
The `Seconds` method is for the `time.Duration` right?

Reply:
If you are unsure then I would leave out this change for now.

Reply:
Yes, the conversion is unnecessary.

Reply:
OK